extensions/amp-story/1.0/amp-story.js

```diff
@@ -347,15 +347,15 @@ export class AmpStory extends AMP.BaseElement {
     // prerendering, because of a height incorrectly set to 0.
     this.mutateElement(() => {});
 
+    const pageId = this.getInitialPageId_();
     if (
       !this.win.CSS?.supports?.('height: 1dvh') &&
       !getStyle(this.win.document.documentElement, '--story-dvh')
     ) {
-      this.getViewport().onResize((size) => this.polyfillDvh_(size));
-      this.polyfillDvh_(this.getViewport().getSize());
+      this.getViewport().onResize((size) => this.polyfillDvh_(size, pageId));
+      this.polyfillDvh_(this.getViewport().getSize(), pageId);
```

Comment:
At this point, some attributes that affect styling (e.g. `[active]` or `data-story-supports-landscape`) are not set. This should go at the very end of buildCallback.

Also, the code could be much easier, e.g.
```
const pageEl = this.element.querySelector('amp-story-page[active]');
this.storeService_.dispatch(Action.SET_PAGE_SIZE, {width, height} = pageEl.getLayoutBox());

```